### PR TITLE
[AIRFLOW-1335] Use MemoryHandler for buffered logging

### PR DESCRIPTION
**Pull request by saguziel**
Dear Airflow maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [x] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "[AIRFLOW-XXX] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-1335

### Description
- [x] Here are some details about my PR, including screenshots of any UI changes: Use buffered logging

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason: Logging statements work

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

**Comment by saguziel** [comment]
@aoen @AllisonWang @bolkedebruin

File: airflow/config_templates/default_airflow.cfg [hunk 1]
```diff
@@ -49,6 +49,13 @@ s3_log_folder =
 # Logging level
 logging_level = INFO
 
+# Size of logging buffer
+logging_buffer_size = 4096
+# Logging level minimum to flush buffer before full
+# See python doc, DEBUG=10,INFO=20,WARN=30,ERROR=40
```
Comment (aoen) [inline airflow/config_templates/default_airflow.cfg]: Would be good to replace "python doc" with an actual link, it's not 100% clear where to look.

**Comment by aoen** [comment]
LGTM

File: airflow/bin/cli.py [hunk 2]
```diff
@@ -90,7 +90,10 @@ def sigquit_handler(sig, frame):
 
 def setup_logging(filename):
     root = logging.getLogger()
-    handler = logging.FileHandler(filename)
+    handler_buffer_size = conf.getInt('core', 'logging_buffer_size')
```
Comment (wmedlar) [inline airflow/bin/cli.py:93]: This should be [`conf.getint`](https://github.com/apache/incubator-airflow/blob/master/airflow/configuration.py#L218).

Comment (wmedlar) [reply airflow/bin/cli.py:93]: Right now it's breaking the webserver cli in `master`.

```python
Traceback (most recent call last):
  File "/usr/local/bin/airflow", line 28, in <module>
    args.func(args)
  File "/usr/local/lib/python3.6/site-packages/airflow/bin/cli.py", line 753, in webserver
    handle = setup_logging(log_file)
  File "/usr/local/lib/python3.6/site-packages/airflow/bin/cli.py", line 93, in setup_logging
    handler_buffer_size = conf.getInt('core', 'logging_buffer_size')
AttributeError: module 'airflow.configuration' has no attribute 'getInt'
```

Comment (saguziel) [reply airflow/bin/cli.py:93]: Shoot. Didn't realize this code path was only hit with `-D`. Ill 'rpobably revert this

**Pull request closed without merging**